drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -1289,10 +1295,11 @@ local function measurementHandlerFactory(capability_name, attribute, target_unit
     end
 
     if value then
-      device:emit_event_for_endpoint(ib.endpoint_id, attribute({value = value, unit = unit_strings[target_unit]}))
       -- handle case where device profile supports both fineDustLevel and dustLevel
       if capability_name == capabilities.fineDustSensor.NAME and device:supports_capability(capabilities.dustSensor) then
```

Comment:
Maybe I'm reading this wrong, but does it need an additional check like this?
```suggestion
      if (capability_name == capabilities.fineDustSensor.NAME and device:supports_capability(capabilities.dustSensor)) or (capability_name == capabilities.dustSensor.NAME and device:supports_capability(capabilities.fineDustSensor)) then
```

Reply:
No, we wouldn't need this check since dustSensor is a superset of fineDustSensor, but not the other way around. So this check handles the edge case that we get a pm25 reading but we don't support fineDustSensor (because we use dustSensor instead)